controllers/managedocs_controller.go

```diff
@@ -885,6 +923,47 @@ func (r *ManagedOCSReconciler) reconcileAlertRelabelConfigSecret() error {
 	return nil
 }
 
+// reconcilePrometheusService function wait for prometheus Service
+// to start and sets appropriate annotation for 'service-ca' controller
+func (r *ManagedOCSReconciler) reconcilePrometheusService() error {
+	r.Log.Info("Reconciling Prometheus Service")
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.prometheusService, func() error {
+		if err := r.own(r.prometheusService); err != nil {
+			return err
+		}
+
+		desired := templates.PrometheusServiceTemplate.DeepCopy()
+		r.prometheusService.Spec.Ports = desired.Spec.Ports
+		r.prometheusService.Spec.Selector = map[string]string{
+			monLabelKey: monLabelValue,
+		}
+		if r.prometheusService.Annotations == nil {
+			r.prometheusService.Annotations = make(map[string]string)
+		}
+		r.prometheusService.Annotations[servingCertSecretBetaAnnotationKey] = prometheusServiceName
+		r.prometheusService.Annotations[servingCertSecretAlphaAnnotationKey] = prometheusServiceName
+		return nil
+	})
+	return err
+}
+
+func (r *ManagedOCSReconciler) reconcileCABundleConfigMap() error {
```

Comment:
Function name is still not fixed 
```suggestion
func (r *ManagedOCSReconciler) reconcileServingCertCABundleConfigMap() error {
```

Reply:
Done